woocommerce/woocommerce
# Add create attribute term modal

### All Submissions:






### Changes proposed in this Pull Request:

This allows users to create a new attribute term if the term they are searching for doesn't exist within the attribute term list.
This is currently dependent on #34999 to be merged, which is why it's pointing at that branch.

https://user-images.githubusercontent.com/2240960/196231848-4b686769-c032-42f2-af67-3508b953140c.mp4

Address the term creation part in #34331  .

### How to test the changes in this Pull Request:

1. Load this branch, build it, and make sure you have the `new-product-management-experience` feature flag enabled (you can do so using the latest version of the Beta tester within the mono repo).
2. Go to **Products > Add new (MVP)** and fill out some of the initial details like name and price
3. Scroll down and click **Add first attribute** in the attributes section
4. Select one of the attributes
5. Search for a attribute term that doesn't exist yet
6. The 'Create "your new attribute term name"' should show as an option
7. Click that and another modal should show with the create fields.
8. Fill out all the fields and click **Add** this should create the attribute term and add it as a selected item within the terms field
9. With `localStorage.setItem( 'debug', 'wc-admin:*' );` run prior, it should show that a `product_attribute_term_add` and `product_attribute_term_add_success` event has been fired.
10. Open the modal again and select **cancel** this shouldn't do anything.
11. A user should not be able to click **add** when the name is removed.

### Other information:




### FOR PR REVIEWER ONLY:
